### DISCUSSION TITLE
qpdf --check tells me PDF is PDF Version: 1.7 extension level 8?

### DISCUSSION THREAD

**Discussion by vxbush** [qpdf/qpdf#704] [General]
I'm using qpdf 64-bit on Win 10 21H2 for the first time, and in doing some testing the qpdf --check command is returning the PDF format as "PDF Version: 1.7 extension level 8", even though the source document is exactly the same as previous testing with the 32-bit version of qpdf.  What is this? My Google searching is not finding any such extension level 8 for PDFs.  Using 10.6.3.

**Comment by jberkenbilt** [comment]
Can you tell me what the problem is? What were you expecting to see? Did the 32-bit version tell you something different? PDF version 1.7, extension level 8 is a valid PDF version. I'd have to research to figure out which Acrobat version in matches, but it's definitely a legitimate version.

**Comment by vxbush** [comment]
The 32-bit version of 10.6.3 gave me simply the string "PDF Version: 1.7" without any indication of the extension. I'm not getting any warnings after the conversion, but I'm trying to be very careful to make sure the new PDFs I create are as close a match to the ones before running them through QPDF. (I'm adding encryption and I have stepped through the process to make sure I am exactly matching the settings for accessibility, printing high/low res, modification permission, etc.)

**Reply by jberkenbilt** [reply]
I can't think of any reason why the 32 bit and 64 bit version should give different results. Are you sure you are looking at the same file? If you add encryption 256-bit encryption, qpdf will set the extension level to 8. This is per spec. If the original file was 1.7 and you added encryption, then you would see a difference, but both versions of the code should give the same results on any given file.

**Reply by vxbush** [reply]
I went back and carefully went over my notes, and I was wrong; the issue was whether I was doing a check on an encrypted file or not. Both the 32- and 64-bit versions of qpdf do the same thing; before encryption, the file is showing up as PDF 1.7; only after encryption am I seeing extension 8 appearing. So that matches with your description of it being an extension 8 issue. My apologies for the mixup.
